m4/check_pkg_gdrcopy.m4

```diff
@@ -0,0 +1,34 @@
+# -*- autoconf -*-
+#
+# Copyright (c) 2025      Amazon.com, Inc. or its affiliates. All rights reserved.
+#
+# See LICENSE.txt for license information
+#
+
+AC_DEFUN([CHECK_PKG_GDRCOPY], [
+  check_pkg_found=yes
+
+  check_pkg_CPPFLAGS_save="${CPPFLAGS}"
+
+  AC_ARG_WITH([gdrcopy],
+     [AS_HELP_STRING([--with-gdrcopy=PATH], [Path to non-standard gdrcopy installation])])
+
+  AS_IF([test -n "${with_gdrcopy}"], [NCCL_NET_OFI_DISTCHCK_CONFIGURE_FLAGS="$NCCL_NET_OFI_DISTCHCK_CONFIGURE_FLAGS --with-gdrcopy=${with_gdrcopy}"])
```

Comment:
I'm not really convinced that we need this.  None of the unit tests should fail if gdrcopy isn't built.